app/controllers/api/v2/moves_actions.rb

```diff
@@ -13,18 +13,16 @@ def show_and_render
 
     def create_and_render
       move = Move.new(move_attributes)
-      move.determine_supplier
-      authorize!(:create, move)
-      move.save!
-
+      move.determine_supplier(doorkeeper_application_owner)
+      move.save! # NB: the move should be saved and notified even if the current user is no longer authorised to view it
```

Comment:
Hmm... I'm not sure this is the correct rule to apply. If we're now assuming on creation that the doorkeeper supplier token has priority then when a supplier creates the move we would always have a token. So this doesn't need to be changed from the current implementation.

My understanding is that moves created in the front end will always be mapped to a supplier (via SupplierLocation), and moves created by suppliers will always have a supplier API token. So the existing logic should work. Also there is another story in flight to ensure that a move have a supplier before being created - that I think is a better fix to ensuring data integrity. See: https://github.com/ministryofjustice/hmpps-book-secure-move-api/pull/878

The potential behaviour here of creating a move but returning a 403 means that clients are likely to retry the same operation repeatedly meaning we could end up with garbage data.